src/install.rs

```diff
@@ -55,15 +55,17 @@ pub fn install(
         fs::rename(&archive_file, &executable_path)?;
     } else {
         if !cfg!(target_os = "macos") {
-			archive::unpack(&archive_file, &destination, true)?;
-        }            
+            archive::unpack(&archive_file, &destination, true)?;
+        }
     }
 
     if cfg!(target_os = "macos") {
         if archive_file.to_string_lossy().contains("zip") {
             mac_process_zip(&archive_file);
         } else {
-            mac_process_dmg(&archive_file);
+            if let Err(e) = mac_process_dmg(&archive_file) {
+                error!("Mac DMG postprocessing failed. {}", e);
```

Comment:
Thinking about it now, any of these errors should return instead of just logging and carrying on. This would add a failed instance the the UI.

Another option would be to return an error with this message - you can use the `anyhow!` macro for that. The lazier approach would be to just propagate using `?`.

Reply:
Code rewritten.